eo-maven-plugin/src/main/java/org/eolang/maven/tojos/PlacedTojosCached.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2023 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.tojos;
+
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.cactoos.scalar.Sticky;
+import org.cactoos.scalar.Unchecked;
+
+/**
+ * Placed tojos that cached in memory.
+ * After implementing the <a href="https://github.com/yegor256/tojos/issues/60">issue</a>
+ * the class will be removed.
+ * @since 0.30
+ */
+public final class PlacedTojosCached {
+
+    /**
+     * Delegate.
+     */
+    private final PlacedTojos origin;
+
+    /**
+     * Cached placed tojos.
+     */
+    private final Unchecked<Map<String, PlacedTojo>> cache;
+
+    /**
+     * Ctor.
+     * @param tojos Placed tojos.
+     */
+    public PlacedTojosCached(final PlacedTojos tojos) {
+        this.origin = tojos;
+        this.cache = new Unchecked<>(new Sticky<>(() -> PlacedTojosCached.placedCache(tojos)));
+    }
+
+    /**
+     * Find placed tojo by path.
+     * @param target Path.
+     * @return Placed tojo.
+     */
+    public Optional<PlacedTojo> find(final Path target) {
```

Comment:
@mximp I would like to respectfully suggest that the topic we are discussing may be disputable and perhaps it would be best if we discussed it separately in a different issue, if you are interested.

Regarding the use of `Optional`, my intention is to reduce the number of indents and simplify the methods' complexity. For example, consider the following comparison between two methods:
```java
void method(){
  if(cache.has(something){
    val v = cache.get(something);
    doWhatYouWant(v);
  }
}
```
Versus:
```
void method(){
  // cache.get returns Optional
  cache.get(something).ifPresent(this::doWhatYouWant);
}  
```
The latter approach results in fewer indents, less complexity, and fewer lines of code. However, I acknowledge that Optional can be misused as well, as shown in this example:
```java
void method(){
  Optional<V> ov = cache.get(something);
  if(ov.isPresent()){
    val v = ov.get();
    doWhatYouWant(v);
  }
}
```
Nevertheless, even in this case, we can avoid creating a `has` method.